src/bigquery/bigquery.py

```diff
@@ -0,0 +1,142 @@
+# Copyright © 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Dharmendra G Patel <dhpatel@redhat.com>
+#
+"""Bigquery implementation to read big data for manifest files."""
+import time
+from rudra import logger
+from rudra.data_store.aws import AmazonS3
+from src.config.settings import AWS_SETTINGS
+from src.bigquery.base import BigqueryBuilder, DataProcessing
+from src.bigquery.base_collector import BaseCollector
+from src.bigquery.maven_collector import MavenCollector
+from src.bigquery.npm_collector import NpmCollector
+from src.bigquery.pypi_collector import PypiCollector
+
+ECOSYSTEM_MANIFEST_MAP = {
+    'maven': 'pom.xml',
+    'npm': 'package.json',
+    'pypi': 'requirements.txt',
+}
+
+
+class Bigquery(BigqueryBuilder):
+    """Base big query class."""
+
+    def __init__(self):
+        """Initialize MavenBigQuery object."""
+        super().__init__()
+        self.query_job_config.use_legacy_sql = False
+        self.query_job_config.use_query_cache = True
+        self.query = """
+            SELECT con.content AS content, L.path AS path
+            FROM `bigquery-public-data.github_repos.contents` AS con
+            INNER JOIN (
+                SELECT files.id AS id, files.path as path
+                FROM `bigquery-public-data.github_repos.languages` AS langs
+                INNER JOIN `bigquery-public-data.github_repos.files` AS files
+                ON files.repo_name = langs.repo_name
+                    WHERE (
+                        (
+                            REGEXP_CONTAINS(TO_JSON_STRING(language), r'(?i)java') AND
+                            files.path LIKE '%{m}'
+                        ) OR
+                        (
+                            REGEXP_CONTAINS(TO_JSON_STRING(language), r'(?i)python') AND
+                            files.path LIKE '%{p}'
+                        ) OR
+                        (
+                            files.path LIKE '%{n}'
+                        )
+                    )
+            ) AS L
+            ON con.id = L.id;
+        """.format(m=ECOSYSTEM_MANIFEST_MAP['maven'],
+                   p=ECOSYSTEM_MANIFEST_MAP['pypi'],
+                   n=ECOSYSTEM_MANIFEST_MAP['npm'])
+
+
+class BQDataProcessing(DataProcessing):
```

Comment:
There are 3 parts to DataProcessing
1) querying
2) Collecting
3) Persistence to S3

Can the functionality be split like this and the processor only acts like co-ordinator?

Reply:
It is of that form, BigQuery == Querying, BQDataProcessing == Collecting and DataProcessing == Persistence. I will rename all this class as suggested.